hts-precompile/IHederaTokenService.sol

```diff
@@ -441,6 +569,189 @@ interface IHederaTokenService {
+    /// @param account The account address to be unfrozen
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function unFreezeToken(address token, address account)
+        external
+        returns (int responseCode);
+
+    /// Operation to grant kyc to token account
+    /// @param token The token address
+    /// @param account The account address to grant kyc
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function grantTokenKyc(address token, address account)
+        external
+        returns (int responseCode);
+
+    /// Operation to revoke kyc to token account
+    /// @param token The token address
+    /// @param account The account address to revoke kyc
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function revokeTokenKyc(address token, address account)
+        external
+        returns (int responseCode);
+
+    /// Operation to pause token
+    /// @param token The token address to be paused
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function pauseToken(address token) external returns (int responseCode);
+
+    /// Operation to unpause token
+    /// @param token The token address to be unpaused
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function unPauseToken(address token) external returns (int responseCode);
+
+    /// Operation to wipe fungible tokens from account
+    /// @param token The token address
+    /// @param account The account address to revoke kyc
+    /// @param amount The number of tokens to wipe
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function wipeTokenAccount(
+        address token,
+        address account,
+        uint32 amount
+    ) external returns (int responseCode, uint32 wipedAmount);
+
+    /// Operation to wipe non fungible tokens from account
+    /// @param token The token address
+    /// @param account The account address to revoke kyc
+    /// @param  serialNumbers The serial numbers of token to wipe
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function wipeTokenAccount(
+        address token,
+        address account,
+        uint64[] memory serialNumbers
+    ) external returns (int responseCode, uint64[] memory wipedSerialNumbers);
+
+    /// Operation to update token info
+    /// @param token The token address
+    /// @param tokenInfo The hedera token info to update token with
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    /// @return newTokenInfo The updated hedera token info
+    function updateTokenInfo(address token, HederaToken memory tokenInfo)
+        external
+        returns (int responseCode, HederaToken memory newTokenInfo);
```

Comment:
I think that `newTokenInfo` would use unnecessary resources both here and in the nodes to encode for the return --- if `responseCode` == SUCCESS, all of the fields in the `tokenInfo` argument will be updated. If not, nothing will be updated. Isn't a response code sufficient?

Reply:
I'm in agreement. Removed